### ISSUE TITLE
[RuntimeError: no running event loop] with runworker in channels 2.3.1 with latest asgiref (3.4.0)

### ISSUE THREAD

**Issue by bj00rn** [django/channels#1713]
Getting the following error after installing django-channels with the runworker command. It's working perfectly with asiref<3.4.0

```
$ python3 manage.py runworker background-task  --settings=config.settings.local

Running worker for channels ['background-task']
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/xxx/.pyenv/versions/cumulus_dev/lib/python3.8/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/Users/xxx/.pyenv/versions/cumulus_dev/lib/python3.8/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/xxx/.pyenv/versions/cumulus_dev/lib/python3.8/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/xxx/.pyenv/versions/cumulus_dev/lib/python3.8/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/Users/xxx/.pyenv/versions/cumulus_dev/lib/python3.8/site-packages/channels/management/commands/runworker.py", line 46, in handle
    worker.run()
  File "/Users/xxx/.pyenv/versions/cumulus_dev/lib/python3.8/site-packages/asgiref/server.py", line 59, in run
    event_loop = get_running_loop()
RuntimeError: no running event loop
```

pip freeze
```
asgiref==3.4.0
channels==2.3.1
channels-redis==2.4.1
daphne==2.4.0
Django==2.2.16
```

MacOS, python 3.8.4, pyenv

**Comment by carltongibson** [comment]
Hi @bj00rn — can you put together a minimal reproduce for this?

**Comment by bj00rn** [comment]
@carltongibson 

Here is a minimal

https://github.com/bj00rn/asgiref-bug/tree/main

**Comment by carltongibson** [comment]
Same issue also reported as https://github.com/django/asgiref/issues/278

**Comment by tsantor** [comment]
I believe I have the same issue...

`docker-compose -f local.yml run --rm django python manage.py runworker game_engine`

```
Running worker for channels ['game_engine']
[2021-07-01 22:00:29,442 INFO/Process-1] [runworker] Running worker for channels ['game_engine']
Traceback (most recent call last):
  File "manage.py", line 31, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 395, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 330, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 371, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.8/site-packages/channels/management/commands/runworker.py", line 46, in handle
    worker.run()
  File "/usr/local/lib/python3.8/site-packages/asgiref/server.py", line 59, in run
    event_loop = get_running_loop()
RuntimeError: no running event loop
make: *** [fom_engine] Error 1
```

**Comment by boonleng** [comment]
Same issue here as I am going through the tutorial of Channels v3 at https://channels.readthedocs.io/en/stable/tutorial/index.html. Using asgiref==3.3.4 is okay.

**Comment by julianwachholz** [comment]
This appears to be a Django 4.0 support blocker now since it requires `asgiref>=3.4.1`

**Comment by bj00rn** [comment]
@carltongibson I understand from @julianwachholz this was fixed in https://github.com/django/asgiref/pull/296

Can we expect a release anytime soon?

This issue is preventing us from upgrading to django 4 since it requires `asgiref>=3.4.1`

Cheers

**Comment by carltongibson** [comment]
@bj00rn I have several personal matters to attend to before getting back to this currently, so not imminent. 

If it's **just** the `asgiref` version, and you (or someone) can get everything ready, I am happy to push a release. 
(Same applies if it's more than that too...) 

Otherwise you'll need to wait a small while yet I'm afraid.

**Comment by bj00rn** [comment]
> @bj00rn I have several personal matters to attend to before getting back to this currently, so not imminent.
> 
> If it's **just** the `asgiref` version, and you (or someone) can get everything ready, I am happy to push a release. (Same applies if it's more than that too...)
> 
> Otherwise you'll need to wait a small while yet I'm afraid.

@carltongibson I understand. I was thinking, given that the issue has been resolved in asgiref (currently unreleased), we just need a new release of asgiref to get things working again. In hindsight, I should probably have commented on https://github.com/django/asgiref/pull/296 instead, to make things clearer.

**Comment by czlee** [comment]
For what it's worth, I might be missing something, but I'm not finding that 
```
pip install git+https://github.com/django/asgiref@main channels==3.0.4 Django==4.0
```
resolves this issue? (That's an attempt to install the unreleased django/asgiref#296, though with Django 4.0 and the current version of channels.) Did that work for other people?

**Comment by lbt** [comment]
@czlee try running pip uninstall asgiref first.
I had the same issue. pip installing different versions of asgiref didn't make a difference unless I uninstalled first.

Having done that pip install git+https://github.com/django/asgiref@main worked fine for me

**Comment by czlee** [comment]
@lbt Oh, that worked! Interesting that that made a difference, thanks!

**Comment by bj00rn** [comment]
🚀 🚀 🚀 

it seems the issue is resolved in asgiref 3.5.0, closing issue
